src/VisualStudio/Core/Def/Implementation/Snippets/AbstractSnippetInfoService.cs

```diff
@@ -51,18 +52,26 @@ internal abstract class AbstractSnippetInfoService : ForegroundThreadAffinitized
             IAsynchronousOperationListenerProvider listenerProvider)
             : base(threadingContext)
         {
-            AssertIsForeground();
-
             if (serviceProvider != null)
             {
-                var textManager = (IVsTextManager2)serviceProvider.GetService(typeof(SVsTextManager));
-                if (textManager.GetExpansionManager(out _expansionManager) == VSConstants.S_OK)
-                {
-                    ComEventSink.Advise<IVsExpansionEvents>(_expansionManager, this);
-                    _waiter = listenerProvider.GetListener(FeatureAttribute.Snippets);
-                    _languageGuidForSnippets = languageGuidForSnippets;
-                    PopulateSnippetCaches();
-                }
+                _waiter = listenerProvider.GetListener(FeatureAttribute.Snippets);
+                _languageGuidForSnippets = languageGuidForSnippets;
+
+                InitializeAndPopulateSnippetsCacheAsync(threadingContext, serviceProvider).Forget();
+            }
+        }
+
+        private async Task InitializeAndPopulateSnippetsCacheAsync(IThreadingContext threadingContext, Shell.SVsServiceProvider serviceProvider)
+        {
+            var token = _waiter.BeginAsyncOperation(GetType().Name + ".Start");
+
+            await threadingContext.JoinableTaskFactory.SwitchToMainThreadAsync();
+
+            var textManager = (IVsTextManager2)serviceProvider.GetService(typeof(SVsTextManager));
```

Comment:
```suggestion
            var textManager = (IVsTextManager2)await serviceProvider.GetServiceAsync(typeof(SVsTextManager));
            Assumes.Present(textManager);
```

Reply:
📝 You'll need to change the parameter type to `Shell.IAsyncServiceProvider` for this one.

Reply:
IVsTextManager is not free threaded and casting requires the UI thread - is it still advisable to use GetServiceAsync in this scenario?  Will GetServiceAsync run on the UI thread?

Reply:
We will still switch to the UI thread, but `GetServiceAsync` ensures we will not be blamed if the initialization for the package that provides `SVsTextManager` is slow.